snapcraft/plugins/go.py

```diff
@@ -39,6 +39,10 @@
       This entry tells the checked out `source` to live within a certain path
       within `GOPATH`.
       This is not needed and does not affect `go-packages`.
+
+    - go-buildtags:
+      (string)
```

Comment:
So:

go-buildtags: [tag1, tag2]

And that will turn into -tags=tag1,tag2. lgtm, I'll make the change.